// @acts/front-core/dist/components/ngx-loading-spinner/loading-spinners.module.js
import { NgModule } from '@angular/core';
import { LoadingSpinnerComponent } from './loading-spinner.component';
import * as i0 from "@angular/core";
var LoadingSpinnersModule = /** @class */ (function () {
    function LoadingSpinnersModule() {
    }
    LoadingSpinnersModule.ɵfac = function LoadingSpinnersModule_Factory(t) { return new (t || LoadingSpinnersModule)(); };
    LoadingSpinnersModule.ɵmod = i0.ɵɵdefineNgModule({ type: LoadingSpinnersModule });
    LoadingSpinnersModule.ɵinj = i0.ɵɵdefineInjector({});
    return LoadingSpinnersModule;
}());
export { LoadingSpinnersModule };
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(LoadingSpinnersModule, { declarations: [LoadingSpinnerComponent], exports: [LoadingSpinnerComponent] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(LoadingSpinnersModule, [{
        type: NgModule,
        args: [{
                declarations: [LoadingSpinnerComponent],
                exports: [LoadingSpinnerComponent]
            }]
    }], null, null); })();
