// @acts/front-core/dist/components/ngx-loading-spinner/loading-spinners.service.js
import { Injectable } from '@angular/core';
import { share } from 'rxjs/operators';
import { Observable } from 'rxjs';
import * as i0 from "@angular/core";
var LoadingSpinnersService = /** @class */ (function () {
    function LoadingSpinnersService() {
        var _this = this;
        this.loaderObservable = new Observable(function (observer) {
            _this.loaderObserver = observer;
        }).pipe(share());
    }
    LoadingSpinnersService.prototype.show = function (name) {
        if (this.loaderObserver) {
            this.loaderObserver.next({ name: name, show: true });
        }
    };
    LoadingSpinnersService.prototype.hide = function (name) {
        if (this.loaderObserver) {
            this.loaderObserver.next({ name: name, show: false });
        }
    };
    LoadingSpinnersService.prototype.getUniqueName = function () {
        var random = function () {
            return Math.floor((1 + Math.random()) * 0x10000)
                .toString(16)
                .substring(1);
        };
        return random() + random() + '-' + random() + '-' + random() + '-' + random() + '-' + new Date().valueOf();
    };
    Object.defineProperty(LoadingSpinnersService.prototype, "defaultSpinnerFilePath", {
        get: function () {
            return this._defaultSpinnerFilePath;
        },
        set: function (filePath) {
            this._defaultSpinnerFilePath = filePath;
        },
        enumerable: false,
        configurable: true
    });
    LoadingSpinnersService.ɵfac = function LoadingSpinnersService_Factory(t) { return new (t || LoadingSpinnersService)(); };
    LoadingSpinnersService.ɵprov = i0.ɵɵdefineInjectable({ token: LoadingSpinnersService, factory: LoadingSpinnersService.ɵfac, providedIn: 'root' });
    return LoadingSpinnersService;
}());
export { LoadingSpinnersService };
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(LoadingSpinnersService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
