// @acts/front-core/dist/core/core-state/core.model.js
/**
 * createCore
 *
 * @export
 * @param {Partial<Core>} { loading = false }
 * @returns
 */
export function createCore(_a) {
    var _b = _a.loading, loading = _b === void 0 ? false : _b;
    return {
        loading: loading,
        loadingType: 'default',
        theme: undefined,
        system: {
            config: {
                host: undefined
            },
            maintenance: {
                maintenanceFlag: undefined,
                title: undefined,
                message: undefined
            },
            version: {
                latestVersion: undefined,
                minimumVersion: undefined,
                title: undefined,
                message: undefined,
                // eslint-disable-next-line @typescript-eslint/camelcase
                dlUrl_ios: undefined,
                // eslint-disable-next-line @typescript-eslint/camelcase
                dlUrl_and: undefined
            }
        },
        userConfig: [
            {
                display: undefined,
                functionId: undefined
            }
        ]
    };
}
