// @acts/front-core/dist/declarations/directive/event-hook.directive.js
import { Directive, HostListener, Input } from '@angular/core';
import { LoggerService } from '../../core/logger/logger.service';
import * as i0 from "@angular/core";
import * as i1 from "../../core/logger/logger.service";
/**
 * hook click event
 *
 * @export
 * @class EventLogDirective
 */
var EventLogDirective = /** @class */ (function () {
    /**
     * Creates an instance of EventLogDirective.
     * @param {LoggerService} logger
     * @memberof EventLogDirective
     */
    function EventLogDirective(logger) {
        this.logger = logger;
    }
    /**
     * hook click event
     *
     * @param {*} eventName
     * @memberof EventLogDirective
     */
    EventLogDirective.prototype.onClick = function () {
        this.logger.tapEvent(this.appEventLog);
    };
    EventLogDirective.ɵfac = function EventLogDirective_Factory(t) { return new (t || EventLogDirective)(i0.ɵɵdirectiveInject(i1.LoggerService)); };
    EventLogDirective.ɵdir = i0.ɵɵdefineDirective({ type: EventLogDirective, selectors: [["", "appEventLog", ""]], hostBindings: function EventLogDirective_HostBindings(rf, ctx) { if (rf & 1) {
            i0.ɵɵlistener("click", function EventLogDirective_click_HostBindingHandler() { return ctx.onClick(); });
        } }, inputs: { appEventLog: "appEventLog" } });
    return EventLogDirective;
}());
export { EventLogDirective };
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(EventLogDirective, [{
        type: Directive,
        args: [{
                selector: '[appEventLog]'
            }]
    }], function () { return [{ type: i1.LoggerService }]; }, { appEventLog: [{
            type: Input,
            args: ['appEventLog']
        }], onClick: [{
            type: HostListener,
            args: ['click']
        }] }); })();
