// @acts/front-core/dist/core/core.module.js
import { NgModule } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';
import { AuthQuery } from './auth/auth.query';
import { AuthService } from './auth/auth.service';
import { CoreFirebaseService } from './core-state/core-firebase.service';
import { CoreQuery } from './core-state/core.query';
import { CoreService } from './core-state/core.service';
import { CoreEnvironmentManager } from './core.environment';
import { GoogleTagManagerService } from './google-tag-manager/google-tag-manager.service';
import { DefaultHttpInterceptor } from './http-interceptors/default-http-interceptor';
import { CodeLabelUtil } from './utils/codelabel.util';
import { ThemeUtil } from './utils/theme.util';
import { TimerUtil } from './utils/timer.util';
import { CommonValidator } from './validators/common.validator';
import { NumberValidator } from './validators/number.validator';
import { StringValidator } from './validators/string.validator';
import * as i0 from "@angular/core";
/**
 * Containers Core NgModule
 *
 * @export
 * @class CoreModule
 */
var CoreModule = /** @class */ (function () {
    function CoreModule() {
    }
    CoreModule.ɵfac = function CoreModule_Factory(t) { return new (t || CoreModule)(); };
    CoreModule.ɵmod = i0.ɵɵdefineNgModule({ type: CoreModule });
    CoreModule.ɵinj = i0.ɵɵdefineInjector({ providers: [
            AuthService,
            AuthQuery,
            CoreService,
            CoreFirebaseService,
            CoreQuery,
            TimerUtil,
            ThemeUtil,
            DefaultHttpInterceptor,
            TranslateModule,
            CoreEnvironmentManager,
            CodeLabelUtil,
            NumberValidator,
            StringValidator,
            CommonValidator,
            GoogleTagManagerService
        ], imports: [[]] });
    return CoreModule;
}());
export { CoreModule };
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(CoreModule, [{
        type: NgModule,
        args: [{
                providers: [
                    AuthService,
                    AuthQuery,
                    CoreService,
                    CoreFirebaseService,
                    CoreQuery,
                    TimerUtil,
                    ThemeUtil,
                    DefaultHttpInterceptor,
                    TranslateModule,
                    CoreEnvironmentManager,
                    CodeLabelUtil,
                    NumberValidator,
                    StringValidator,
                    CommonValidator,
                    GoogleTagManagerService
                ],
                imports: [],
                entryComponents: []
            }]
    }], null, null); })();
