// @acts/front-core/dist/declarations/core.declarations.module.js
import { NgModule } from '@angular/core';
import { EventLogDirective } from './directive/event-hook.directive';
import { DateToJapaneseYearPipe } from './pipe/date-to-japanese-year.pipe';
import { Nl2BrPipe } from './pipe/nl2br.pipe';
import * as i0 from "@angular/core";
/**
 * Containers Core NgModule
 *
 * @export
 * @class CoreModule
 */
var CoreDeclarationsModule = /** @class */ (function () {
    function CoreDeclarationsModule() {
    }
    CoreDeclarationsModule.ɵfac = function CoreDeclarationsModule_Factory(t) { return new (t || CoreDeclarationsModule)(); };
    CoreDeclarationsModule.ɵmod = i0.ɵɵdefineNgModule({ type: CoreDeclarationsModule });
    CoreDeclarationsModule.ɵinj = i0.ɵɵdefineInjector({});
    return CoreDeclarationsModule;
}());
export { CoreDeclarationsModule };
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(CoreDeclarationsModule, { declarations: [EventLogDirective, Nl2BrPipe, DateToJapaneseYearPipe], exports: [EventLogDirective, Nl2BrPipe, DateToJapaneseYearPipe] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(CoreDeclarationsModule, [{
        type: NgModule,
        args: [{
                exports: [EventLogDirective, Nl2BrPipe, DateToJapaneseYearPipe],
                declarations: [EventLogDirective, Nl2BrPipe, DateToJapaneseYearPipe],
                entryComponents: []
            }]
    }], null, null); })();
